tests/core/test_integration.py

```diff
@@ -797,6 +808,173 @@ def test_ignored_snapshots(sushi_context: Context):
     assert exp.to_table("sushi__dev.order_items") in metadata.qualified_views
 
 
+@pytest.mark.integration
+@pytest.mark.core_integration
+def test_scd_type_2(tmp_path: pathlib.Path):
+    def create_source_dataframe(values: t.List[t.Tuple[int, int, str, str]]) -> pd.DataFrame:
+        return pd.DataFrame(
+            np.array(
+                values,
+                [
+                    ("id1", "int32"),
+                    ("id2", "int32"),
+                    ("name", "object"),
+                    ("updated_at", "datetime64[ns]"),
+                ],
+            ),
+        )
+
+    def create_target_dataframe(
+        values: t.List[t.Tuple[int, int, str, str, str, t.Optional[str]]]
+    ) -> pd.DataFrame:
+        return pd.DataFrame(
+            np.array(
+                values,
+                [
+                    ("id1", "int32"),
+                    ("id2", "int32"),
+                    ("name", "object"),
+                    ("updated_at", "datetime64[ns]"),
+                    ("valid_from", "datetime64[ns]"),
+                    ("valid_to", "datetime64[ns]"),
+                ],
+            ),
+        )
+
+    def replace_source_table(
+        context: Context,
+        values: t.List[t.Tuple[int, int, str, str]],
+    ):
+        df = create_source_dataframe(values)
+        context.engine_adapter.replace_query(
+            "scd_type_2_input",
+            df,
+            columns_to_types={
+                "id1": exp.DataType.build("int"),
+                "id2": exp.DataType.build("int"),
+                "name": exp.DataType.build("STRING"),
+                "updated_at": exp.DataType.build("TIMESTAMP"),
+            },
+        )
+
+    def compare_dataframes(df1: pd.DataFrame, df2: pd.DataFrame):
+        df1 = df1.sort_values(by=["id1", "id2"]).reset_index(drop=True)
+        df2 = df2.sort_values(by=["id1", "id2"]).reset_index(drop=True)
+        pd.testing.assert_frame_equal(df1, df2)
+
+    shutil.copy(str(pathlib.Path("examples/sushi/config.py")), str(tmp_path / "config.py"))
+    (tmp_path / "models").mkdir()
+    with open(tmp_path / "models" / "scd_type_2.sql", "w+") as f:
+        f.write(
+            """
+MODEL (
```

Comment:
I initially converted `sushi.items` to scd type 2 (it currently it similar and does a daily snapshot history) but I found that we had a bunch of tests written assuming it was a incremental model and I didn't want to go in and update all of those tests. Really the key concern with scd type 2 is how it handles the variety of updates it gets (add, updates, deletes) which is covered by this test. It didn't seem high enough return to actual change `sushi.items` for the time it would take to do given what can be covered with just this test.

Reply:
make a new model, don't change a current one

Reply:
I don't want to pollute the data model with a new scd type 2 just for the sake of having it in there, especially when we already have a model that should be a scd type 2 (`sushi.items`). Can you explain the issue with what is here? Is it the fact that we don't have an example in the examples folder? If so I am intending on fixing this once Trey is done creating the final examples and we are able to isolate these current examples to just tests.

Reply:
i don't understand the issue, sushi.examples is a test project and thus we can add directly in there all the new types, that's the whole point. i don't like how this is an inline example that's going to be harder to maintain